Add https related configuration for the HAProxy section (#7403)
wazuh/wazuh-documentation: source/user-manual/manager/configuring-cluster/advanced-settings.rst

The helper uses the Dataplane API to communicate with HAProxy and update the configuration according to the changes in the Wazuh cluster.

This is the basic configuration. You need to replace ``<DATAPLANE_USER>`` and ``<DATAPLANE_PASSWORD>`` with the chosen user and password. 

.. tabs::

   .. group-tab:: HTTP

      .. code-block:: yaml
         :emphasize-lines: 8,9

         dataplaneapi:
            host: 0.0.0.0
            port: 5555
            transaction:
                transaction_dir: /tmp/haproxy
            user:
            - insecure: true
               password: <DATAPLANE_PASSWORD>
               name: <DATAPLANE_USER>
         haproxy:
            config_file: /etc/haproxy/haproxy.cfg
            haproxy_bin: /usr/sbin/haproxy
            reload:
               reload_delay: 5
               reload_cmd: service haproxy reload
               restart_cmd: service haproxy restart

   .. group-tab:: HTTPS

      .. note::

>          If you use HTTPS as the Dataplane API communication protocol, you need to add the ``tls`` field and related subfields, being ``tls_port``, ``tls_certificate`` and ``tls_key`` mandatory for the communication. The ``tls_ca`` field is only necessary when using Client Side Certificates.

Reviewer: ```suggestion
         If you use HTTPS as the Dataplane API communication protocol, it is mandatory to set the ``tls`` field and related subfields ``tls_port``, ``tls_certificate`` and ``tls_key`` in the configuration. The ``tls_ca`` field is only necessary when using Client Side Certificates.
```

         
         To generate the certificate files both for the HAProxy instance and the Wazuh server, the following command should be used:

         .. code-block:: console

            # openssl req -x509 -newkey rsa:4096 -keyout <KEY_FILE_NAME> -out <CERTIFICATE_FILE_NAME> -sha256 -nodes -addext "subjectAltName=DNS:<FQDN>" -subj "/C=US/ST=CA/O=Wazuh>/CN=<CommonName>"

      .. code-block:: yaml
         :emphasize-lines: 2,3,8,9,10,11,12,15,16

         dataplaneapi:
            scheme: 
            - https
            host: 0.0.0.0
            port: 5555
            transaction:
               transaction_dir: /tmp/haproxy
            tls:
               tls_port: 6443
               tls_certificate: /etc/haproxy/ssl/<HAPROXY_CERTIFICATE_FILE>
               tls_key: /etc/haproxy/ssl/<HAPROXY_CERTIFICATE_KEY_FILE>
               tls_ca: /etc/haproxy/ssl/<CLIENT_SIDE_CERTIFICATE_FILE>
            user:

>             - insecure: true

Author: This modification raises the following: `configuration error: configuration file /etc/haproxy/dataplaneapi.yml not valid (only yaml format is supported): yaml: line 14: did not find expected '-' indicator`
The suggested configuration follows the alignment mentioned in the [official Dataplane repo example](https://github.com/haproxytech/dataplaneapi/blob/e2389be328814290ef8e92ad4e3242d75e6a1a00/configuration/examples/example-full.yaml#L34-L37).

            </nodes>
            <hidden>no</hidden>
            <disabled>no</disabled>
            <haproxy_helper>
               <haproxy_disabled>no</haproxy_disabled>
               <haproxy_address><HAPROXY_ADDRESS></haproxy_address>
               <haproxy_user><DATAPLANE_USER></haproxy_user>
               <haproxy_password><DATAPLANE_PASSWORD></haproxy_password>
               <haproxy_protocol>https</haproxy_protocol>
               <haproxy_port>6443</haproxy_port>
               <haproxy_cert><HAPROXY_CERTIFICATE_FILE></haproxy_cert>
               <client_cert><CLIENT_SIDE_CERTIFICATE_FILE></client_cert>
               <client_cert_key><CLIENT_SIDE_CERTIFICATE_KEY_FILE></client_cert_key>
            </haproxy_helper>
         </cluster>

      Where:

      -  :ref:`haproxy_disabled <haproxy_disabled>`: Indicates whether the helper is disabled or not in the master node.
      -  :ref:`haproxy_address <haproxy_address>`: IP or DNS address to connect with HAProxy.
      -  :ref:`haproxy_user <haproxy_user>`: Username to authenticate with HAProxy.
      -  :ref:`haproxy_password <haproxy_password>`: Password to authenticate with HAProxy.

>       -  :ref:`haproxy_protocol <haproxy_protocol>`: HTTPS protocol to use for the HAProxy Dataplane API communication.

Author: Changing the last sentence to `It is recommended to be set to https` since `http` is also allowed.

>       -  :ref:`haproxy_protocol <haproxy_protocol>`: HTTPS protocol to use for the HAProxy Dataplane API communication.

Reviewer: I used `must` because this is the HTTPS tab, so in order to use an encrypted connection, setting HTTP doesn't work
